spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/FeatureManager.java

```diff
@@ -144,7 +145,11 @@ public void putAll(Map<? extends String, ? extends Object> m) {
         if (m == null) {
             return;
         }
-        
+
+        if (m.size() == 1 && m.get("featureManagement") != null) {
+            m = (Map<? extends String, ? extends Object>) m.get("featureManagement");
+        }
+
         for (String key : m.keySet()) {
             addToFeatures(m, key, "");
         }
```

Comment:
I see that we are changing the variable method value, setting a new value. This practice can turn the readability sometimes really confusing.  What do you think about it?

```java
    @Override
    public void putAll(Map<? extends String, ? extends Object> m) {
        if (m == null) {
            return;
        }

        Set<? extends String> featureKeys = getFeatureManagementKeysOrKeySet(m);
        for (String key : featureKeys) {
            addToFeatures(m, key, "");
        }
    }

    private Set<? extends String> getFeatureManagementKeysOrKeySet(Map<? extends String, ?> m) {
        if (isOnlyFeatureManagement(m)) {
            return ((Map<? extends String, ? extends Object>) m.get("featureManagement")).keySet();
        }

        return m.keySet();
    }

    private boolean isOnlyFeatureManagement(Map<? extends String, ?> m) {
        return m.size() == 1 && m.containsKey("featureManagement");
    }
```